src/core_modules/capture-core/components/WidgetReferral/ReferralActions/ReferralActions.component.js

```diff
@@ -34,29 +40,47 @@ const styles = () => ({
     },
 });
 
-export const ReferralActionsPlain = ({ classes, type }: Props) => {
+export const ReferralActionsPlain = ({
+    classes,
+    type,
+    selectedType,
+    ...passOnProps
+}: Props) => {
     const [selectedAction, setSelectedAction] = React.useState();
-    return (<DataSection
-        dataTest="referral-section"
-        sectionName={i18n.t('Referral actions')}
-    >
-        <div className={classes.wrapper}>
-            {type === referralStatus.REFERRABLE ? Object.keys(mainOptionTranslatedTexts).map(key => (
-                <Radio
-                    key={key}
-                    name={`referral-action-${key}`}
-                    checked={key === selectedAction}
-                    label={mainOptionTranslatedTexts[key]}
-                    onChange={(e: Object) => setSelectedAction(e.value)}
-                    value={key}
+    const { programStage } = useProgramStageInfo(selectedType.toConstraint.programStage.id);
```

Comment:
Hey,
I get an error for a relationship without `toConstraint.programStage`. Maybe you can add a `?` to fix it. Something like this: `useProgramStageInfo(selectedType.toConstraint.programStage?.id);`
Thanks!

Reply:
Good catch! This was never meant to be hardcoded to the `toConstraint`-object either way, but the other constraint regardless of `from` or `to`. Should be fixed now.